images/router/haproxy/conf/haproxy-config.template

```diff
@@ -199,9 +243,22 @@ frontend fe_no_sni
 
   # map to http backend
   # Search from most specific to general path (host case).
+  acl edge_http_backend base,map_beg(/var/lib/haproxy/conf/os_edge_http_be.map) -m found
+  use_backend be_edge_http_%[base,map_beg(/var/lib/haproxy/conf/os_edge_http_be.map)] if edge_http_backend
+
+{{ if matchPattern "true|TRUE" (env "ROUTER_ALLOW_WILDCARD_ROUTES" "")}}
```

Comment:
I assumed it would be a wee bit more expensive because its doing a regexp compare so there is some cost to matching against a pattern. But it is in a if block so is only enabled if the router allows wildcard routes. 

The strange part was just did a quick and dirty test with ab and I see different results - removed https out of the mix as there is overhead benchmarking w/ cert checks etc and this is what I see: 

For a `wildcard` route that allows http requests and returns a header dump:
`time ab -n 10000 -c 50  -H "Host: www2.open.header.test" http://127.0.0.1/` 
50% of the requests finished in 13 ms, 95% finished in 19ms, and 99% in 23 ms - longest was 27ms.  Elapsed time = 0m2.777s 

For a non-wildcard route: 
 `time ab -n 10000 -c 50  -H "Host: allow-http.header.test" http://127.0.0.1/`
50% of the requests took the same time 13 ms, 95% took 21 ms , 99% took 25 ms and the longest one took 28ms.  Elapsed time: 0m2.855s
 
So the wildcard checks are actually a wee bit faster thanks to a compiled regexp. Of course the performance test is not a real-world test but it still does have some merit and the results are not to say the least interesting.